codecs/qoi/enc/qoi_enc.cpp

```diff
@@ -0,0 +1,34 @@
+#include <emscripten/bind.h>
+#include <emscripten/val.h>
+
+#define QOI_IMPLEMENTATION
+#include "qoi.h"
+
+using namespace emscripten;
+
+struct QoiOptions {};
+
+thread_local const val Uint8Array = val::global("Uint8Array");
+
+val encode(std::string buffer, int width, int height, QoiOptions options) {
+  int compressedSizeInBytes;
+  qoi_desc desc;
+  desc.width = width;
+  desc.height = height;
+  desc.channels = 4;
```

Comment:
We should make `channels` configurable via `options` (`3` and `4` are valid values)! If there is no transparency (i.e. 3 channels), the resulting image file should get smaller. 

While we are at it, let’s make `colorspace` configurable as well.

Reply:
As @jakearchibald said on chat: It’d be idea if the code would automatically use 3-channel mode if there are no transparent pixels.

Reply:
Turns out a side-effect of QOI's simplicity is that it can't encode RGB data differently from RGBA. Pretty much just sets the alpha to `255` and the source image's channels matter only during decoding. So yeah, using only 3 channels ends up with the same compressed size as if you use all 4 channels (for opaque images).

Reply:
You are right. I read trhe source wrong. Ignore me ;D